s2a_options.go

```diff
@@ -119,6 +121,8 @@ type ServerOptions struct {
 	LocalIdentities []Identity
 	// S2AAddress is the address of the S2A.
 	S2AAddress string
+	// V2 is a flag that enables S2Av2.
+	V2 bool
```

Comment:
The V2 integration tests are separate from the s2a_e2e_test.go, in s2av2_e2e_test.go. Since I added the V2 option to the s2a ClientOptions and ServerOptions, should I update the existing s2a_e2e_test.go with V2 enabled in the ClientOptions / ServerOptions? Or would it be redundant to have the same e2e tests in both places?

Reply:
Can we also add new test cases to s2a_e2e_test.go? The benefit of having these new test cases (in addition to s2av2_e2e_test.go) is that it lets us know that the top-level APIs are working are expected, which I think we need (even if there is a lot of overlap with s2av2_e2e_test.go).

Reply:
Will do!